.buildkite/scripts/build_image.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common.sh
+
+trap docker_logout EXIT
```

Comment:
Instead of using `trap` for the `docker_logout` we could use a `pre-exit`[[1](https://buildkite.com/docs/agent/v3/hooks#job-lifecycle-hooks)] hook that would take care of that, similar to doing the `docker login`.
@mrodm what do you think?

Reply:
I like that approach of using Buildkite hooks. :+1:

Reply:
It's also a good idea, but unfortunately, it also doesn't work.  I got the error `docker: command not found`. This error confirms my assumption about the dedicated environment for hooks. If it's true we have to run the `docker logout` command in the environment where we ran the `docker login` command i.e. directly in the agent